gslb/cache/controller_obj_cache.go

```diff
@@ -571,10 +581,33 @@ func GetDetailsFromAviGSLB(gslbSvcMap map[string]interface{}) (uint32, []GSMembe
 				weight = 0
 			}
 			weightI := int32(weight)
-			ipList = append(ipList, ipAddr+"-"+strconv.Itoa(int(weightI)))
+			vsUUID, ok := member["vs_uuid"].(string)
+			if !ok {
+				gslbutils.Warnf("couldn't parse the vs uuid, assigning \"\": %v", member)
+				vsUUID = ""
+			}
+			var syncType int
+			if vsUUID == "" {
+				syncType = gslbutils.SyncTypeThirdPartyVips
+			}
+			controllerUUID, ok := member["cluster_uuid"].(string)
+			if !ok {
+				gslbutils.Warnf("couldn't parse the controller cluster uuid, assigning \"\": %v", member)
+				controllerUUID = ""
```

Comment:
Should we change syncType here ?
In general I was wondering what should be done if controllerUUID is empty either in Avi obj cache or in the object annotation.

Reply:
As discussed, we shouldn't change the syncType here. Since, we want a deterministic way of handling the sync type and we depend on the user input for that. Also, the sync type has to be persisted between the restarts and we find that using the gs object.